tutorials/marble/version-control.md

```diff
@@ -0,0 +1,354 @@
+# Version Control
+
+Version control is a useful tool in any workflow.  It allows you to work on copies of your work and make changes to them without affecting the original.
+
+
+## Table of Contents
+- [Introduction to Git](#intro-git)
+- [Git Best Practices](#best-practices)
+- [Setting up a Git Repository](#setup-git)
+  - [Creating a GitHub Account](#github-account)
+  - [Creating a Remote Repository](#remote-repository)
+  - [Creating a Personal Access Token](#personal-access-token)
+  - [Setting up using the Graphical User Interface](#setup-git-gui)
+    - [Adding a Remote Repository](#add-remote-gui)
+    - [Creating a Local Repository](#local-repository-gui)
+  - [Setting up using the Terminal](#setup-git-terminal)
+    - [Adding a Remote Repository](#add-remote-terminal)
+    - [Creating a Local Repository](#local-repository-terminal)
+- [Using Git](#using-git)
+  - [Using Git with the Graphical User Interface](#using-git-gui)
+    - [Making Your First Commit](#first-commit)
+    - [Staging a Commit](#stage-gui)
+    - [Making a Commit](#commit-gui)
+    - [Pushing a Commit](#push-gui)
+    - [Creating a Branch](#create-branch-gui)
+  - [Using Git with the Terminal](#using-git-terminal)
+    - [Staging a Commit](#stage-terminal)
+    - [View Staged Files](#view-stage-terminal)
+    - [Making a Commit](#commit-terminal)
+    - [Pushing a Commit](#push-terminal)
+    - [Creating a Branch](#create-branch-terminal)
+- [Creating a Pull Request](#create-pr)
+- [Merging Your Branch](#merge-branch)
+
+## <a id="intro-git"></a>Introduction to Git
+
+Git is one of the most popular version control systems in use today. It has a native commandline interface but also has 
+a desktop GUI client for Windows and MacOS.  The online GitHub website allows you to store your repository online and access it anywhere with a free account. 
+
+## <a id="best-practices"></a>Git Best Practices
+
+### Organize with Branches
+
+Create a branch for each issue you're working on.  Have one for data ingestion, one for visualization, and one for bug fixes.
+
+This keeps the workflow clean.  If something went wrong you know which branch the changes came from.
+
+### Write Descriptive Commit Messages
+
+A good commit message documents the changes made in the commit so that when looking through the commits you can see the direction your work is going in.
+
+### Commit finished tasks only
+
+Each commit should be a completed logical chunk of the overall workflow.  If you need to commit often, split up the workflow into smaller tasks.
+
+### Test code before committing
+
+Related to only committing finished tasks, test your code to ensure it does what it is intended to do before committing.
```

Comment:
minor formatting issue, consider moving the section to after these concepts are discussed.

Reply:
I think you're right. Moved this to the bottom.